pkg/controller/job/job_controller.go

```diff
@@ -1120,6 +1132,20 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 			jobCtx.finishedCondition = newFailedConditionForFailureTarget(jobCtx.finishedCondition, jm.clock.Now())
 		}
 	}
+	if feature.DefaultFeatureGate.Enabled(features.JobSuccessPolicy) {
```

Comment:
The `func successCriteriaMetCondition` checks if the Job already has a `SuccessCriteriaMet`  condition, and the `func isSuccessCriteriaMetCondition` checks if the type of the condition (batch.JobCondition) is `SuccessCriteriaMet` conditoin.

We need to verify if the type of the `finishedCondition` is `SuccessCriteriaMet` here: 
https://github.com/kubernetes/kubernetes/blob/18795c121cf0ade6c82be76ebf6cbfb0346ab032/pkg/controller/job/job_controller.go#L1165-L1166

Then, to avoid adding the `SuccessCriteriaMet` condition twice or more, we need to verify if the Job doesn't have a `SuccessCriterieMet` condition here: https://github.com/kubernetes/kubernetes/blob/18795c121cf0ade6c82be76ebf6cbfb0346ab032/pkg/controller/job/job_controller.go#L1168-L1171

But, I agree with available of potential replacement of https://github.com/kubernetes/kubernetes/blob/18795c121cf0ade6c82be76ebf6cbfb0346ab032/pkg/controller/job/job_controller.go#L1165-L1166 with 

```golang
isSuccessCriteriaMetCondition(jobCtx.finishedCondition)
```

@soltysh Does it make sense?

Reply:
> The `func successCriteriaMetCondition` checks if the Job already has a `SuccessCriteriaMet` condition, and the `func isSuccessCriteriaMetCondition` checks if the type of the condition (batch.JobCondition) is `SuccessCriteriaMet` conditoin.
> 
> We need to verify if the type of the `finishedCondition` is `SuccessCriteriaMet` here:
> 
> https://github.com/kubernetes/kubernetes/blob/18795c121cf0ade6c82be76ebf6cbfb0346ab032/pkg/controller/job/job_controller.go#L1165-L1166
> 
> Then, to avoid adding the `SuccessCriteriaMet` condition twice or more, we need to verify if the Job doesn't have a `SuccessCriterieMet` condition here:
> 
> https://github.com/kubernetes/kubernetes/blob/18795c121cf0ade6c82be76ebf6cbfb0346ab032/pkg/controller/job/job_controller.go#L1168-L1171
> 
> But, I agree with available of potential replacement of
> 
> https://github.com/kubernetes/kubernetes/blob/18795c121cf0ade6c82be76ebf6cbfb0346ab032/pkg/controller/job/job_controller.go#L1165-L1166
> 
> with
> ```go
> isSuccessCriteriaMetCondition(jobCtx.finishedCondition)
> ```
> 
> @soltysh Does it make sense?

@soltysh NVM, my understanding isn't correct. We can remove this.
ref: https://github.com/kubernetes/kubernetes/pull/123412#discussion_r1514419691